src/javascript/app_2/App/Components/Form/DatePicker/date_picker.jsx

```diff
@@ -5,22 +5,36 @@ import { IconArrow }   from 'Assets/Common';
 import {
     formatDate,
     daysFromTodayTo }  from 'Utils/Date';
+import Calendar        from 'App/Components/Elements/Calendar';
 import DatePickerInput from './date_picker_input.jsx';
-import Calendar        from '../../Elements/Calendar';
 
 class DatePicker extends React.PureComponent {
     state = {
         value                : '',
         is_datepicker_visible: false,
         is_close_btn_visible : false,
+        calendar_date        : '',
+        selected_date        : '',
     };
 
-    componentWillReceiveProps({ value, mode }) {
-        if (value === this.state.value) return;
-        this.updateDatePickerValue(value, mode);
+    static getDerivedStateFromProps({ value, mode, ...props }, state) {
+        const { date_format, start_date } = props;
+        const new_date = (mode === 'duration') ? moment.utc().add(value, 'days').format(date_format) : value;
+        if ((moment.utc(new_date, date_format).isValid() || !new_date)) {
+            if (!new_date) {
+                const current_date = moment.utc(start_date).format(date_format);
+                state.calendar_date = current_date;
+                state.selected_date = current_date;
+            } else {
+                state.selected_date = formatDate(new_date);
+                state.calendar_date = formatDate(new_date);
+            }
+        }
```

Comment:
can we use `this.updateDatePickerValue(value, mode)` here as well?

Reply:
We cannot use that, we do not have access to `this` in that method.